### PR TITLE
IgnoreNullValues obsolete

### LINKED ISSUE #24471: System.Text.Json IgnoreNullValues not shown
image

I'm using Visual Studio 2019 16.8.6 with .Net 5.0.103 SDK

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #24471

### CODE REVIEW

File: docs/standard/serialization/system-text-json-ignore-properties.md [hunk 1]
```diff
@@ -121,6 +121,9 @@ Here's an example object to serialize and JSON output:
 }
 ```
 
+> [!NOTE]
+> The <xref:System.Text.Json.JsonSerializerOptions.IgnoreNullValues> property is obsolete in .NET 5 and later versions, so it isn't shown by IntelliSense. For the current way to ignore null values, see [how to ignore all null-value properties in .NET 5 and later](system-text-json-ignore-properties.md?pivots=dotnet-5-0#ignore-all-null-value-properties).
```
Comment (gewarren) [inline docs/standard/serialization/system-text-json-ignore-properties.md:125]:
```suggestion
> The <xref:System.Text.Json.JsonSerializerOptions.IgnoreNullValues> property is obsolete in .NET 6 and later versions, so it isn't shown by IntelliSense. For the current way to ignore null values, see [how to ignore all null-value properties in .NET 6 and later](system-text-json-ignore-properties.md?pivots=dotnet-5-0#ignore-all-null-value-properties).
```

File: docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md [hunk 2]
```diff
@@ -290,7 +290,7 @@ The `Newtonsoft.Json` `[JsonConstructor]` attribute lets you specify which const
 <xref:System.Text.Json> in .NET Core 3.1 provides the following ways to ignore properties while serializing:
 
 * The [[JsonIgnore]](system-text-json-ignore-properties.md#ignore-individual-properties) attribute on a property causes the property to be omitted from the JSON during serialization.
-* The [IgnoreNullValues](system-text-json-ignore-properties.md#ignore-all-null-value-properties) global option lets you ignore all null-value properties.
+* The [IgnoreNullValues](system-text-json-ignore-properties.md#ignore-all-null-value-properties) global option lets you ignore all null-value properties. `IgnoreNullValues` is obsolete in .NET 5 and later versions, so it isn't shown by IntelliSense. For the current way to ignore null values, see [how to ignore all null-value properties in .NET 5 and later](system-text-json-ignore-properties.md?pivots=dotnet-5-0#ignore-all-null-value-properties).
```
Comment (gewarren) [inline docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md:293]:
```suggestion
* The [IgnoreNullValues](system-text-json-ignore-properties.md#ignore-all-null-value-properties) global option lets you ignore all null-value properties. `IgnoreNullValues` is obsolete in .NET 6 and later versions, so it isn't shown by IntelliSense. For the current way to ignore null values, see [how to ignore all null-value properties in .NET 6 and later](system-text-json-ignore-properties.md?pivots=dotnet-5-0#ignore-all-null-value-properties).
```

**Review by gewarren — COMMENTED** [review]
It's not obsolete in .NET 5 - so I think you also need to update system-text-json-ignore-properties.md. Also, maybe we should update the Workarounds section in [this article](https://docs.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0020) to point to "how to ignore all null-value properties in .NET 6 and later"?

**Comment by tdykstra** [comment]
>It's not obsolete in .NET 5 

Deprecated would be the word for .NET 5, as the new way to do the same thing was introduced in .NET 5 and we want people to stop using the old way in .NET 5 and later.

File: docs/standard/serialization/system-text-json-ignore-properties.md [hunk 3]
```diff
@@ -121,6 +121,9 @@ Here's an example object to serialize and JSON output:
 }
 ```
 
+> [!NOTE]
+> The <xref:System.Text.Json.JsonSerializerOptions.IgnoreNullValues> property is deprecated in .NET 5 and later versions, so it isn't shown by IntelliSense. For the current way to ignore null values, see [how to ignore all null-value properties in .NET 5 and later](system-text-json-ignore-properties.md?pivots=dotnet-5-0#ignore-all-null-value-properties).
```
Comment (gewarren) [inline docs/standard/serialization/system-text-json-ignore-properties.md:125]:
```suggestion
> The <xref:System.Text.Json.JsonSerializerOptions.IgnoreNullValues> property is deprecated in .NET 5 and later versions. For the current way to ignore null values, see [how to ignore all null-value properties in .NET 5 and later](system-text-json-ignore-properties.md?pivots=dotnet-5-0#ignore-all-null-value-properties).
```

I assume only obsolete types aren't shown by IntelliSense? And that's cool - good for IntelliSense for not showing those.

**Review by gewarren — APPROVED** [review]

**Comment by tdykstra** [comment]
Thanks for the review @gewarren, good catch on the obsolete vs. deprecated terminology!

**Pull request merged**